fossunited/api/profile.py

```diff
@@ -78,17 +77,23 @@ def update_profile(fields_dict):
             "mastodon": fields_dict.get("mastodon"),
         }
 
-        frappe.db.set_value(USER_PROFILE, user_doc.name, updated_fields)
+        profile = frappe.get_doc(USER_PROFILE, user_doc.name)
+        for field, value in updated_fields.items():
+            if hasattr(profile, field):
+                setattr(profile, field, value)
+        profile.save(ignore_permissions=True)
 
         user_updates = {}
         if fields_dict.get("full_name") != user_doc.full_name:
-            user_updates["full_name"] = fields_dict.get("full_name")
+            user_updates["first_name"] = fields_dict.get("full_name").split(" ")[0]
 
         if fields_dict.get("username") != user_doc.username:
             user_updates["username"] = fields_dict.get("username")
 
         if user_updates:
-            frappe.db.set_value("User", user_doc.user, user_updates)
+            user = frappe.get_doc("User", user_doc.user)
+            for field, value in user_updates.items():
+                setattr(user, field, value)
```

Comment:
don't we need to call `user.save(...)` here like we do with `profile.save(...)` above? the `setattr` will only modify the `user` in memory but not in the database, right?

Reply:
Fixed this. It is indeed needed

Reply:
it will be good to add unittests to catch these issues. if you don't have bandwidth at the moment, please open an issue regarding adding unittests for these functions/functionality